apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/QueryUtils.java

```diff
@@ -178,4 +221,59 @@ public static boolean isInRange(Instant lastUpdated, DateRangeParam range) {
     }
     return true;
   }
+
+  /**
+   * Convert an integer value into claims data available BitSet.
+   *
+   * @param maskVal integer value suitable for decomposing into a Claims BitSet.
+   * @return {@link BitSet} denoting which claims have data.
+   */
+  public static BitSet convertClaimsBtimaskValue(int maskVal) {
+    BitSet rslt = new BitSet(maskVal);
+    rslt.set(CARRIER_HAS_DATA, (maskVal & V_CARRIER_HAS_DATA) != 0);
+    rslt.set(INPATIENT_HAS_DATA, (maskVal & V_INPATIENT_HAS_DATA) != 0);
+    rslt.set(OUTPATIENT_HAS_DATA, (maskVal & V_OUTPATIENT_HAS_DATA) != 0);
+    rslt.set(SNF_HAS_DATA, (maskVal & V_SNF_HAS_DATA) != 0);
+    rslt.set(DME_HAS_DATA, (maskVal & V_DME_HAS_DATA) != 0);
+    rslt.set(HHA_HAS_DATA, (maskVal & V_HHA_HAS_DATA) != 0);
+    rslt.set(HOSPICE_HAS_DATA, (maskVal & V_HOSPICE_HAS_DATA) != 0);
+    rslt.set(PART_D_HAS_DATA, (maskVal & V_PART_D_HAS_DATA) != 0);
+    return rslt;
+  }
+
+  /**
+   * Query database to determine which claim types have data for the specified beneficiary.
+   *
+   * @param entityManager {@link EntityManager} used to query database.
+   * @param bene_id used to identify the Benficiary to chek claims for.
+   * @return {@link BitSet} denoting which claims have data.
+   */
+  public static BitSet hasClaimsData(EntityManager entityManager, long bene_id) {
+    /*
+     * execute a database function that returns a bitwise mask value that denotes that the given
+     * claim type will have data for the specified beneficiayrId. This represents fast and efficient
```

Comment:
`beneficiayrId` -> `bene_id or beneficiaryId`

Reply:
well yes and no; beneficiaryId is a function of some really old code that was specific to the various ORM models. At one point, Karl wanted to get rid of those Java field names (that he actually created) when we did a big schema migration a while back. We never did that follow-up work, but I tend to use CCW name(s) unless dealing with an ORM model.

Reply:
Ah I was more so suggesting it because of Java naming conventions.

Reply:
+1 here also.  parameter names are meant to be camel case.

Reply:
changed to camel case.